test/integration/apiserver/flowcontrol/concurrency_util_test.go

```diff
@@ -0,0 +1,496 @@
+	noxu1PLSeatUtilAvg := (laterPLSeatUtilSum[priorityLevelNoxu1.Name] - earlierPLSeatUtilSum[priorityLevelNoxu1.Name]) / float64(laterPLSeatUtilCount[priorityLevelNoxu1.Name]-earlierPLSeatUtilCount[priorityLevelNoxu1.Name])
+	noxu2PLSeatUtilAvg := (laterPLSeatUtilSum[priorityLevelNoxu2.Name] - earlierPLSeatUtilSum[priorityLevelNoxu2.Name]) / float64(laterPLSeatUtilCount[priorityLevelNoxu2.Name]-earlierPLSeatUtilCount[priorityLevelNoxu2.Name])
+	t.Logf("\nnoxu1RequestExecutionSecondsAvg %v\nnoxu2RequestExecutionSecondsAvg %v", noxu1RequestExecutionSecondsAvg, noxu2RequestExecutionSecondsAvg)
+	t.Logf("\nnoxu1PLSeatUtilAvg %v\nnoxu2PLSeatUtilAvg %v", noxu1PLSeatUtilAvg, noxu2PLSeatUtilAvg)
+
+	wg.Wait() // wait till the client goroutines finish before computing the statistics
+	noxu1ClientRequestLatencySecondsAvg, noxu1ClientRequestLatencySecondsSdev := computeClientRequestLatencyStats(noxu1ClientRequestLatencyCount, noxu1ClientRequestLatencySum, noxu1ClientRequestLatencySumSq)
+	noxu2ClientRequestLatencySecondsAvg, noxu2ClientRequestLatencySecondsSdev := computeClientRequestLatencyStats(noxu2ClientRequestLatencyCount, noxu2ClientRequestLatencySum, noxu2ClientRequestLatencySumSq)
+	t.Logf("\nnoxu1ClientRequestLatencyCount %v\nnoxu2ClientRequestLatencyCount %v", noxu1ClientRequestLatencyCount, noxu2ClientRequestLatencyCount)
+	t.Logf("\nnoxu1ClientRequestLatencySecondsAvg %v\nnoxu2ClientRequestLatencySecondsAvg %v", noxu1ClientRequestLatencySecondsAvg, noxu2ClientRequestLatencySecondsAvg)
+	t.Logf("\nnoxu1ClientRequestLatencySecondsSdev %v\nnoxu2ClientRequestLatencySecondsSdev %v", noxu1ClientRequestLatencySecondsSdev, noxu2ClientRequestLatencySecondsSdev)
+	allDispatchedReqCounts, rejectedReqCounts, err := getRequestCountOfPriorityLevel(loopbackClient)
+	if err != nil {
+		t.Error(err)
+	}
+	t.Logf("\nnoxu1APFRequestCount %v\nnoxu2APFRequestCount %v", allDispatchedReqCounts[priorityLevelNoxu1.Name], allDispatchedReqCounts[priorityLevelNoxu2.Name])
+	if rejectedReqCounts[priorityLevelNoxu1.Name] > 0 {
+		t.Errorf(`%v requests from the "elephant" stream were rejected unexpectedly`, rejectedReqCounts[priorityLevelNoxu2.Name])
+	}
+	if rejectedReqCounts[priorityLevelNoxu2.Name] > 0 {
+		t.Errorf(`%v requests from the "mouse" stream were rejected unexpectedly`, rejectedReqCounts[priorityLevelNoxu2.Name])
+	}
+
+	// Calculate server-side observed concurrency
+	noxu1ObservedConcurrency := noxu1PLSeatUtilAvg * float64(availableSeats[priorityLevelNoxu1.Name])
+	noxu2ObservedConcurrency := noxu2PLSeatUtilAvg * float64(availableSeats[priorityLevelNoxu2.Name])
+	// Expected concurrency is derived from equal throughput assumption on both the client-side and the server-side
+	// Expected concurrency computed can sometimes be larger than the number of available seats. We use the number of available seats as an upper bound
```

Comment:
I'm very suspicious about fixing this via `Min`, if we graph this function it has a sharp inflection point which makes me suspicious of all other values too, not just the ones that went over the limit. I think we should not do this min and make sure the math tolerates large expectations, or find a better way to compute it.

Reply:
(I'd also like to see these variable names shortened, and/or temporaries computed, and/or a helper function called rather than repeat everything)

Reply:
The test on line 137, making sure that availableSeats is larger than 4, ensures that this Min will have no effect for Noxu2.  The Min is here for Noxu2 only for the sake of uniformity; I would be OK with removing this Min.

For Noxu1, we have seen this Min make only a small difference.  The measurements that feed into this are not perfectly aligned (that is impossible Prometheus metrics) and have some other noise in them; the purpose of this Min is to cut out some divergence that comes from the noise.  Really, the comparisons that use these values are only checking that physics works as expected and these requests are not spending time in some totally unexpected way (recall that these checks are derived from equating throughput observed at two places, and that equality is not really optional).  @lavalamp, what graph did you have in mind?  I see three inputs that are not constant (`noxu1RequestExecutionSecondsAvg`, `noxu1ClientRequestLatencySecondsAvg`, and `availableSeats[priorityLevelNoxu1.Name]`).

Reply:
Let me put it another way--

I expect comparisons of data that is noisy in these ways to have some bounded amount of fuzziness, but using min or max is effectively changing the observed data an arbitrary amount. I'd much rather compute the number as it happens, even if it is logically impossible for it to be that high, and tune the comparison(s) to have enough fudge factor to compensate.

Because if it's like 2% more than the max, that's fine, but if it is 200% more, that's a test bug and I want to know about it. Using min obscures the difference.

Reply:
I am collecting some data by removing the min and will report them here soon

Reply:
@MikeSpreitzer @lavalamp 
I removed the Min and collected data from 50 tests and plotted the error, defined by `math.Abs(expected-observed)/expected`, and their ratio to the margin, defined by  `margin := 2 * math.Min(noxu1LatStats.cv, noxu2LatStats.cv)` (cv is [coefficient of variance](https://en.wikipedia.org/wiki/Coefficient_of_variation)). All 50 tests passed. The plots are as follows
![client-latency-error](https://user-images.githubusercontent.com/7364402/182455984-b1e9f745-a349-4dfe-ace9-cbd646256433.png)
![client-latency-error-to-margin-ratio](https://user-images.githubusercontent.com/7364402/182456008-118e06db-6fbb-4f85-8370-deb92e45a277.png)

From the results it seems that there is some room for larger noise, since the error-to-margin ration is less than 0.25 in my experiments. 

Attaching also the logs
[Archive.zip](https://github.com/kubernetes/kubernetes/files/9245073/Archive.zip)